engine/src/engine.rs

```diff
@@ -311,8 +312,13 @@ struct StackExecutorParams {
 
 impl StackExecutorParams {
     fn new(gas_limit: u64, current_account_id: AccountId) -> Self {
+        let random_seed = sdk::random_seed().to_fixed_bytes().to_vec();
```

Comment:
Again here I think it is better to take the random seed as an argument. We want to push sdk access all the way out to `lib.rs`, or if it must happen in core logic then only through one of the sdk traits (in this case I think `sdk::env::Env` makes sense if you are going to use a trait).

Reply:
Got the idea now. Hopefully this time I got it right. To make sure we don't have this problem in the future, we should consider changing visibility from `sdk` to not be used at this point.